network/src/legacy/gossip/mod.rs

```diff
@@ -334,11 +334,19 @@ pub fn register_validator<C: ChainContext + 'static, S: NetworkSpecialization<Bl
 	let gossip_side = validator.clone();
 	let gossip_engine = sc_network_gossip::GossipEngine::new(
 		service.clone(),
-		executor,
 		POLKADOT_ENGINE_ID,
 		gossip_side,
 	);
 
+	// Ideally this would not be spawned as an orphaned task, but polled by
+	// `RegisteredMessageValidator` which in turn would be polled by a `ValidationNetwork`.
```

Comment:
note that when we remove the legacy code we will use `protocol.rs` which could run this in a future.